### PR TITLE
feat: apply search operation in UserTaskFilter

### LINKED ISSUE #23990: Define filter operations for `UserTaskFilter`
## Description

In order to support advanced search queries for `/user-tasks` we need to prepare the internal structure to support different operations and execute them properly.

## Acceptance criteria

- Extend the required `UserTaskFilter ` properties with operations.
- Make sure underlying structures work with the new structure.
  - `SearchQueryRequestMapper`, `*FilterTransformer`, RDBMS mappers etc.

```[tasklist]
### Tasks
- [ ] https://github.com/camunda/camunda/pull/24166
```

### PR DESCRIPTION
**Pull request by danielkelemen**
## Description

- Applies search operations in UserTaskFilter.

## Related issues

closes https://github.com/camunda/camunda/issues/23990

### CODE REVIEW

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java [hunk 1]
```diff
@@ -81,22 +83,11 @@ public SearchQuery toSearchQuery(final UserTaskFilter filter) {
               not(hasChildQuery("taskVariable", taskVarNameQuery)));
 
       // Combine taskVariable, processVariable, and subprocessVariable queries with OR logic
-      variablesQuery = or(taskVariableQuery, processVariableCondition);
+      queries.add(or(taskVariableQuery, processVariableCondition));
     }
+    ofNullable(exists("flowNodeInstanceId")).ifPresent(queries::add); // Default to task
```
Comment (tmetzke) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java:88]: 🔧 I think you can add this search condition directly, right? Without the nullable part of it.
```suggestion
    queries.add(exists("flowNodeInstanceId")); // Default to task
```

**Review by ralfpuchert — COMMENTED** [review]
Do we have tests for the changes?

File: search/search-domain/src/main/java/io/camunda/search/filter/UserTaskFilter.java [hunk 2]
```diff
@@ -8,23 +8,26 @@
...
 import static io.camunda.util.CollectionUtil.collectValuesAsList;
 
+import io.camunda.util.FilterUtil;
 import io.camunda.util.ObjectBuilder;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
-public final record UserTaskFilter(
+public record UserTaskFilter(
```
Comment (tmetzke) [inline search/search-domain/src/main/java/io/camunda/search/filter/UserTaskFilter.java:20]: 💭 Same as in https://github.com/camunda/camunda/pull/24172, we might as well add operation awareness to all fields (except `boolean`s) already.

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java [hunk 3]
```diff
@@ -119,16 +110,20 @@ private SearchQuery getUserTaskKeysQuery(final List<Long> userTaskKeys) {
...
-  private SearchQuery getCandidateGroupsQuery(final List<String> candidateGroups) {
-    return stringTerms("candidateGroups", candidateGroups);
+  private List<SearchQuery> getAssigneesQuery(final List<Operation<String>> assignees) {
+    return stringOperations("assignee", assignees);
   }
 
-  private SearchQuery getAssigneesQuery(final List<String> assignee) {
-    return stringTerms("assignee", assignee);
+  private List<SearchQuery> getPrioritiesQuery(final List<Operation<Integer>> priorities) {
+    return intOperations("priority", priorities);
```
Comment (tmetzke) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java:126]: ❓ This is a new filter option that is untested, right? The other fields are already covered by existing tests and don't change so the existing tests should verify that we don't introduce any regressions with this, right?

**Review by tmetzke — APPROVED** [review]
👍 Generally looks fine
🔧 There some minor issues to cover, e.g. adding operations to all fields and adding transformer tests for the priority filter you added.

Pre-approving already since structurally this is fine and could also be followed up on.

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java (hunk 3 above)
Comment (danielkelemen) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java:126]: Yes, it should be tested too. I will add tests.

File: search/search-domain/src/main/java/io/camunda/search/filter/UserTaskFilter.java (hunk 2 above)
Comment (danielkelemen) [reply search/search-domain/src/main/java/io/camunda/search/filter/UserTaskFilter.java:20]: I'll add them in a follow-up ticket.

**Pull request merged**